File: src/interfaces/todo.ts
// Todo interface matching the database model
export interface Todo {
  id: number;
  title: string;
  description?: string;
  isCompleted: boolean;
  createdAt: Date;
  updatedAt?: Date;
  dueDate?: Date;
  priority?: string;
  category?: string;
  userId?: number;
  user?: User;
}

// User interface matching the database model
export interface User {
  id: number;
  username: string;
  email: string;
  firstName?: string;
  lastName?: string;
  createdAt: Date;
  updatedAt?: Date;
  todos?: Todo[];
}

// DTOs for API requests
export interface CreateTodoRequest {
  title: string;
  description?: string;
  dueDate?: Date;
  priority?: string;
  category?: string;
  userId?: number;
}

export interface UpdateTodoRequest {
  id: number;
  title?: string;
  description?: string;
  isCompleted?: boolean;
  dueDate?: Date;
  priority?: string;
  category?: string;
}

export interface CreateUserRequest {
  username: string;
  email: string;
  firstName?: string;
  lastName?: string;
}

export interface UpdateUserRequest {
  id: number;
  username?: string;
  email?: string;
  firstName?: string;
  lastName?: string;
}

File: backend/TodoApi/Models/Todo.cs
using System;
using System.Collections.Generic;

namespace TodoApi.Models;

public partial class Todo
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public bool IsCompleted { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTime? DueDate { get; set; }

    public string? Priority { get; set; }

    public string? Category { get; set; }

    public int? UserId { get; set; }

    public virtual User? User { get; set; }
}